rfc/rfc-0003-intermediate-value-crds.md

```diff
@@ -1,3 +1,140 @@
-[Draft MR](https://github.com/vmware-tanzu/cartographer/pull/169)
+# RFC 0003 Intermediate Value CRDs
 
-[Draft Text](https://github.com/vmware-tanzu/cartographer/blob/rfc-0003-intermediate-value-crds/rfc/rfc-0003-intermediate-value-crds.md)
\ No newline at end of file
+## Summary
+
+Create a CRD type that will be used to store the values emitted by a node in a supply chain.
+These new objects will give operators increased visibility into the what is happening with
+a supply chain.
+
+## Motivation
+
+Kontinue currently passes values from one component of a supply chain to another in memory.
+This is not observable to end users and can make it difficult to reason about the state of
+the supply chain.
+
+E.g.: when component N points to a SourceTemplate, that component implicitly emits a url
+and a revision, which can be consumed by later components in the SupplyChain DAG. Imagine that
+an operator has made a mistake in writing their SourceTemplate and both the urlPath and the
+revisionPath point to the same field in the GitResource. The supply chain will break in some
+strange way (probably kpack will complain that it is misconfigured). If the operator examines
+the GitResource, they will see that it is behaving as expected. By creating an intermediate
+value CRD, the operator would be able to examine the field and directly observe the mistake.
```

Comment:
The operator can check the value interpolated into the kpack Image, where the error is reported.